tests/Common/ComponentsTest.php

```diff
@@ -38,10 +38,39 @@ public function setUp()
         }
     }
 
-    public function testGetComponentDetail()
+    public function componentsProvider()
     {
+        yield 'defaultBranch' => [
+            function (\Keboola\StorageApi\Client $client) {
+                return new \Keboola\StorageApi\Components($client);
+            }
+        ];
+
+        yield 'devBranch' => [
+            function (\Keboola\StorageApi\Client $client) {
+                $providedToken = $client->verifyToken();
+                $devBranch = new \Keboola\StorageApi\DevBranches($client);
+                $branchName = __CLASS__ . '\\' . $this->getName() . '\\' . $providedToken['id'];
+                $this->deleteBranchesByPrefix($devBranch, $branchName);
+                $branch = $devBranch->createBranch($branchName);
+                return new \Keboola\StorageApi\Components($this->getBranchAwareDefaultClient($branch['id']));
+            }
+        ];
+    }
+
+    /**
+     * @dataProvider componentsProvider
+     */
+    public function testGetComponentDetail(callable $getComponents)
```

Comment:
```suggestion
    /**
     * @dataProvider componentsProvider
     */
    public function testGetComponentDetail(Client $client)
```

A pak v kódu, když potřebuješ výhybku, tak `$client instanceof BranchAwareClient`

Reply:
To z nějakého důvodu nejde?

Reply:
Instance Clienta neni v DataProvideru dostupna - viz nize. Proto je potreba predat `$this` a nastavit `devBranch` clienta az za behu.
> All data providers are executed before both the call to the setUpBeforeClass static method and the first call to the setUp method. Because of that you can't access any variables you create there from within a data provider. This is required in order for PHPUnit to be able to compute the total number of tests.
> https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html#:~:text=a%20depending%20test.-,Note,-All%20data%20providers